conda_build/source.py

```diff
@@ -116,16 +118,16 @@ def download_to_cache(cache_folder, recipe_path, source_dict, verbose=False):
             raise RuntimeError(f"Could not download {url}")
 
     hashed = None
-    for tp in ("md5", "sha1", "sha256"):
-        if tp in source_dict:
-            expected_hash = source_dict[tp]
-            hashed = compute_sum(path, tp)
+
+    for hash_type in set(source_dict).intersection(ACCEPTED_HASH_TYPES):
+        if hash_type in source_dict:
+            expected_hash = source_dict[hash_type]
+            hashed = compute_sum(path, hash_type)
             if expected_hash != hashed:
                 rm_rf(path)
                 raise RuntimeError(
-                    f"{tp.upper()} mismatch: '{hashed}' != '{expected_hash}'"
+                    f"{hash_type.upper()} mismatch: '{hashed}' != '{expected_hash}'"
                 )
-            break
```

Comment:
This is interesting, we were only computing the weakest one and ignoring the rest?

Reply:
yeah that's fun.